File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        
        int left = 0;
        int right = numbers.Length-1;

        while(left < right)
        {
            int result = numbers[left] + numbers[right];
            if(result == target)
            {
                return new int [] { left+1 , right +1};
            }
            else if(result > target)
            {
                right--;
            }
            else if(result < target)
            {
                left++;
            }
        }
        return new int [] { 0 , 0 };
    }
}

File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        
        left = 0
        
        right = len(nums)-1
        
        result = -1
        while left <= right:
            middle = (left + right)//2
            if nums[middle]==target :
                result = middle
                break
        #if middle is less than right then right side is sorted
            if nums[middle] < nums[right]:
                if nums[middle] <= target and target <= nums[right]:
                    left = middle + 1
                else:
                    right= middle - 1
            #if middle is greater than left then left side is sorted
            elif nums[middle] >= nums[left]:
                if nums[middle] >= target and target >= nums[left]:
                    right = middle - 1
                else:
                    left = middle + 1
            
        return result